src/jit/codegenarm.cpp

```diff
@@ -1498,7 +1520,76 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
 
     case GT_STOREIND:
         {
-            NYI("GT_STOREIND");
+            GenTree* data = treeNode->gtOp.gtOp2;
+            GenTree* addr = treeNode->gtOp.gtOp1;
+            GCInfo::WriteBarrierForm writeBarrierForm = gcInfo.gcIsWriteBarrierCandidate(treeNode, data);
+            if (writeBarrierForm != GCInfo::WBF_NoBarrier)
+            {
+                // data and addr must be in registers.
+                // Consume both registers so that any copies of interfering
+                // registers are taken care of.
+                genConsumeOperands(treeNode->AsOp());
+                
+                // At this point, we should not have any interference.
+                // That is, 'data' must not be in REG_ARG_0,
+                //  as that is where 'addr' must go.
+                noway_assert(data->gtRegNum != REG_ARG_0);
+
+                // addr goes in REG_ARG_0
+                if (addr->gtRegNum != REG_ARG_0)
+                {
+                    inst_RV_RV(INS_mov, REG_ARG_0, addr->gtRegNum, addr->TypeGet());
+                }
+
+                // data goes in REG_ARG_1
+                if (data->gtRegNum != REG_ARG_1)
+                {
+                    inst_RV_RV(INS_mov, REG_ARG_1, data->gtRegNum, data->TypeGet());
+                }
+
+                genGCWriteBarrier(treeNode, writeBarrierForm);
+            }
+            else
+            {
+                bool reverseOps = ((treeNode->gtFlags & GTF_REVERSE_OPS) != 0);
+                bool dataIsUnary = false;
+                GenTree* nonRMWsrc = nullptr;
+                // We must consume the operands in the proper execution order, 
+                // so that liveness is updated appropriately.
+                if (!reverseOps)
+                {
+                    genConsumeAddress(addr);
+                }
+                if (data->isContained() && !data->OperIsLeaf())
+                {
+                    dataIsUnary = (GenTree::OperIsUnary(data->OperGet()) != 0);
+                    if (!dataIsUnary)
+                    {
+                        nonRMWsrc = data->gtGetOp1();
+                        if (nonRMWsrc->isIndir() && Lowering::IndirsAreEquivalent(nonRMWsrc, treeNode))
+                        {
+                            nonRMWsrc = data->gtGetOp2();
+                        }
+                        genConsumeRegs(nonRMWsrc);
+                    }
+                }
+                else
+                {
+                    genConsumeRegs(data);
+                }
+                if (reverseOps)
+                {
+                    genConsumeAddress(addr);
+                }
+                if (data->isContained() && !data->OperIsLeaf())
+                {
+                    NYI("RMW?");
+                }
+                else
+                {
+                    emit->emitInsMov(ins_Store(targetType), emitTypeSize(treeNode), treeNode);
+                }
+            }
```

Comment:
This is actually from the ARM64 implementation, I suppose that's why the NYI was "RMW?"